src/pages/settings/Payments/AddDebitCardPage.js

```diff
@@ -226,64 +232,44 @@ class DebitCardPage extends Component {
-                                        </TextLink>
-                                    </Text>
-                                </>
-                            )}
-                            {this.state.manualAddress && (
-                                <>
-                                    <ExpensiTextInput
-                                        label={this.props.translate('addDebitCardPage.billingAddress')}
-                                        containerStyles={[styles.mt4]}
-                                        onChangeText={value => this.clearErrorAndSetValue('addressStreet', value)}
-                                        value={this.state.addressStreet}
-                                        errorText={this.getErrorText('addressStreet')}
-                                    />
-                                    <View style={[styles.flexRow, styles.mt4]}>
-                                        <View style={[styles.flex2, styles.mr2]}>
-                                            <ExpensiTextInput
-                                                label={this.props.translate('common.city')}
-                                                onChangeText={value => this.clearErrorAndSetValue('addressCity', value)}
-                                                value={this.state.addressCity}
-                                                errorText={this.getErrorText('addressCity')}
-                                            />
-                                        </View>
-                                        <View style={[styles.flex1]}>
-                                            <StatePicker
-                                                onChange={value => this.clearErrorAndSetValue('addressState', value)}
-                                                value={this.state.addressState}
-                                                hasError={Boolean(this.getErrorText('addressState'))}
-                                            />
-                                        </View>
-                                    </View>
+                            <AddressSearch
+                                label={this.props.translate('addDebitCardPage.billingAddress')}
+                                containerStyles={[styles.mt4]}
+                                value={this.state.addressStreet}
+                                onChange={(values) => {
+                                    const renamedFields = {
+                                        street: 'addressStreet',
+                                        state: 'addressState',
+                                        zipCode: 'addressZipCode',
+                                    };
+                                    _.each(values, (value, inputKey) => {
+                                        if (inputKey === 'city') {
+                                            return;
+                                        }
```

Comment:
Just noticed this, but don't we want to update the `city` field?

Reply:
I saw that the `city` field wasn't properly added in any of these places here:

https://github.com/Expensify/App/blob/main/src/pages/settings/Payments/AddDebitCardPage.js#L51
https://github.com/Expensify/App/blob/87f41cc92617e6cb5a2deefbefc989a20a9bc2e9/src/pages/settings/Payments/AddDebitCardPage.js#L64
https://github.com/Expensify/App/blob/87f41cc92617e6cb5a2deefbefc989a20a9bc2e9/src/pages/settings/Payments/AddDebitCardPage.js#L76

And, most importantly, we didn't really use it when sending the information to the backend here:

https://github.com/Expensify/App/blob/87f41cc92617e6cb5a2deefbefc989a20a9bc2e9/src/libs/actions/PaymentMethods.js#L43-L71

I thought that it was added by mistake when the "manual" switch was added. It made sense to me that it wasn't being sent to the back because I have seen many card forms that don't have a "City" field.

Reply:
ah, not used by the backend hmm OK then thanks!